apiserver/pkg/manager/resource_manager_test.go

```diff
@@ -0,0 +1,82 @@
+package manager
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/ray-project/kuberay/apiserver/pkg/client"
+	api "github.com/ray-project/kuberay/proto/go_client"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubernetesfake "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestPopulateComputeTemplate(t *testing.T) {
+	ctx := context.Background()
+	namespace := "test-namespace"
+	headConfigMapName := "head"
+	workerConfigMapName := "worker"
+
+	headConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      headConfigMapName,
+			Namespace: namespace,
+		},
+		Data: map[string]string{
+			"name":      headConfigMapName,
+			"namespace": namespace,
+			"cpu":       "2",
+			"memory":    "4",
+		},
+	}
+	workerConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      workerConfigMapName,
+			Namespace: namespace,
+		},
+		Data: map[string]string{
+			"name":      workerConfigMapName,
+			"namespace": namespace,
+			"cpu":       "2",
+			"memory":    "4",
+		},
+	}
+
+	clusterSpec := &api.ClusterSpec{
+		HeadGroupSpec: &api.HeadGroupSpec{
+			ComputeTemplate: headConfigMapName,
+		},
+		WorkerGroupSpec: []*api.WorkerGroupSpec{
+			{
+				ComputeTemplate: workerConfigMapName,
+			},
+		},
+	}
+
+	// mock controller
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
```

Comment:
It's a legacy interface, no need to call `Finish` after 1.14: https://pkg.go.dev/github.com/golang/mock/gomock#Controller.Finish